src/lndk_offers.rs

```diff
@@ -1281,4 +1303,50 @@ mod tests {
         };
         assert!(handler.send_payment(payer_mock, params).await.is_err());
     }
+
+    #[test]
+    fn test_handle_invoice_error_existing_payment() {
+        // Create an OfferHandler with a payment ID in active_payments
+        let handler = OfferHandler::default();
+        let payment_id = PaymentId([42; 32]);
+        let nonce = Nonce::try_from(NONCE_BYTES).unwrap();
+
+        // Insert a payment into active_payments
+        {
```

Comment:
Nit: Is there a reason for the brackets here and below?

Reply:
yes, because of the lock. If we don't do it that way, it would hold it until of the test and actually we would find a deadlock